core/vdbe/mod.rs

```diff
@@ -2087,12 +2116,55 @@ impl Program {
                     }
                     state.pc += 1;
                 }
+                Insn::CreateBtree { db, root, flags: _ } => {
+                    if *db > 0 {
+                        // TODO: implement temp datbases
+                        todo!("temp databases not implemented yet");
+                    }
+                    let mut cursor = Box::new(BTreeCursor::new(
+                        pager.clone(),
+                        0,
+                        self.database_header.clone(),
+                    ));
+
+                    let root_page = cursor.create_tree(1);
+                    state.registers[*root] = OwnedValue::Integer(root_page as i64);
+                    state.pc += 1;
+                }
+                Insn::Close { cursor_id } => {
+                    cursors.remove(cursor_id);
+                    state.pc += 1;
+                }
+                Insn::IsNull { src, target_pc } => {
+                    if matches!(state.registers[*src], OwnedValue::Null) {
+                        state.pc = *target_pc;
+                    } else {
+                        state.pc += 1;
+                    }
+                }
+                Insn::ParseSchema {
+                    db: _,
+                    where_clause,
+                } => {
+                    let conn = self.connection.upgrade();
+                    let conn = conn.as_ref().unwrap();
+                    let stmt = conn.prepare(format!(
+                        "SELECT * FROM  sqlite_schema WHERE {}",
+                        where_clause
+                    ))?;
+                    let rows = Rows { stmt };
+                    let mut schema = RefCell::borrow_mut(&conn.schema);
+                    // TODO: This function below is synchronous, make it not async
+                    parse_schema_rows(Some(rows), &mut *schema, conn.pager.io.clone())?;
```

Comment:
`ParseSchema` opcode documentation is as follows:

> Read and parse all entries from the schema table of database P1 that match the WHERE clause P4. .....
This opcode invokes the parser to create a new virtual machine, then runs the new virtual machine. It is thus a re-entrant opcode.

1. I wonder WHY it is like that,
2. I guess we're not implementing it like that. 

In general should we mark somehow when we have divergent implementations of the opcodes wrt. sqlite, what do you think

Reply:
Wait never mind, this does create a new VDBE as well

Reply:
I believe this is as close as I could get to how they do it. I create a new program which reads all rows and parses them with the correct where clause. 
In my opinion we could just remove the parse_schema stuff and creating a new nested program when we know for sure we just inserted the table and can add it in memory. I don't know why they do it like that, maybe because of some multi processing stuff or whatever but since we have a single process we can simplify a lot of stuff (not in this case, maybe in the future)

Reply:
And to answer my question - it is like that precisely because it creates a new statement

Reply:
Yeah I was mainly just brainfarting, carry on